ayatana-settings

```diff
@@ -45,14 +45,26 @@ class AyatanaSettings:
             sys.exit(1)
 
         buildApp(self)
+        pSource = Gio.SettingsSchemaSource.get_default ()
+        pSchema = pSource.lookup ('org.gnome.desktop.interface', False);
 
-        self.pGnomeSettings = Gio.Settings.new ('org.gnome.desktop.interface')
-        self.pGnomeSettings.connect ('changed::color-scheme', self.onColorSchemeChanged)
-        self.onColorSchemeChanged (self.pGnomeSettings, 'color-scheme')
+        if pSchema:
+
+            bFound = pSchema.has_key ('color-scheme')
+
+            if bFound:
+
+                self.pGnomeSettings = Gio.Settings.new ('org.gnome.desktop.interface')
+                self.pGnomeSettings.connect ('changed::color-scheme', self.onColorSchemeChanged)
+                self.onColorSchemeChanged (self.pGnomeSettings, 'color-scheme')
+
+            else:
+
+                logger.warning ('org.gnome.desktop.interface::color-scheme not found. The application will not react to native dark/light mode changes.')
```

Comment:
I'd rather silently ignore color-scheme not being present. That is: hide the setting in the UI and don't have to bother with the setting not working.

Reply:
There is nothing to hide. This code automatically switches the application to dark/light mode whenever the gschema key changes. This is a must-have for all Gtk 3.24+ applications. The dconf-editor itself is a good example.

Reply:
I have removed the warning, though.